services/checks/checks.go

```diff
@@ -138,22 +139,12 @@ func (s *Service) Run(ctx context.Context) {
 	defer ticker.Stop()
 
 	for {
-		var sttEnabled bool
-		settings, err := models.GetSettings(s.db)
-		if err != nil {
-			s.l.Error(err)
-		}
-		if settings != nil && settings.SaaS.STTEnabled {
-			sttEnabled = true
-		}
-
-		if sttEnabled {
-			nCtx, cancel := context.WithTimeout(ctx, checksTimeout)
-			s.collectChecks(nCtx)
-			s.executeChecks(nCtx)
-			cancel()
-		} else {
-			s.l.Info("STT is not enabled, doing nothing.")
+		if err := s.StartChecks(ctx); err != nil {
+			if errors.Is(err, services.ErrSTTDisabled) {
```

Comment:
Sorry, we don't use errors.Is/As in PMM yet: there are differences between it and `errors.Cause`. Use the later function or just compare with `==`. And let's blacklist a linter/rule that suggests otherwise.

Reply:
Done